zookeeper/src/test/java/com/linecorp/armeria/client/endpoint/zookeeper/ZookeeperEndpointGroupTest.java

```diff
@@ -0,0 +1,127 @@
+    private static final ArrayList<Endpoint> initializedEndpointGroupList = new ArrayList<>();
+
+    @Override
+    public ZKInstance instance() {
+        return zkInstance;
+    }
+
+    @BeforeClass
+    public static void start() {
+        initializedEndpointGroupList.add(Endpoint.of("127.0.0.1", 1234, 2));
+        initializedEndpointGroupList.add(Endpoint.of("127.0.0.1", 2345, 4));
+        initializedEndpointGroupList.add(Endpoint.of("127.0.0.1", 3456, 2));
+        try {
+            zkInstance.start().result(duration);
+        } catch (Throwable throwable) {
+            throwable.printStackTrace();
+        }
+    }
+
+    @AfterClass
+    public static void stop() {
+        try {
+            zkInstance.stop().ready(duration);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Before
+    public void connectZk() {
+        createOrUpdateZNode(endpointListToByteArray(initializedEndpointGroupList));
+        //forcing a get on the Option as we should be connected at this stage
+        zookeeperEndpointGroup = new ZookeeperEndpointGroup(
+                zkInstance.connectString().get(), zNode, 3000,
+                new DefaultZkNodeValueConverter());
+    }
+
+    @After
+    public void disconnectZk() {
+        try {
+            zookeeperEndpointGroup.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testUpdateEndpointGroup() {
+        List<Endpoint> expected = ImmutableList.of(Endpoint.of("127.0.0.1", 8001, 2),
+                                                   Endpoint.of("127.0.0.1", 8002, 3));
+        createOrUpdateZNode(endpointListToByteArray(expected));
+        assertEquals(expected, zookeeperEndpointGroup.endpoints());
+    }
+
+    @Test
+    public void testGetEndpointGroup() {
+        assertEquals(initializedEndpointGroupList, zookeeperEndpointGroup.endpoints());
+    }
+
+    private void createOrUpdateZNode(byte[] nodeValue) {
+        zkInstance.connect().map(closeableZooKeeper -> {
+            if (closeableZooKeeper.exists(zNode).get()) {
+                return closeableZooKeeper.setData(zNode, nodeValue,
+                                                  closeableZooKeeper.exists(zNode,
+                                                                            false).getVersion());
+            }
+            return closeableZooKeeper.create(zNode, nodeValue, ZooDefs.Ids.OPEN_ACL_UNSAFE,
+                                             CreateMode.PERSISTENT);
+
+        });
+        assertExists(zNode);
+    }
+
+    private static byte[] endpointListToByteArray(List<Endpoint> endpointList) {
+        return endpointList.stream().map(
+                endPointA -> (endPointA.authority() + ':' + endPointA.weight())
+        ).reduce(
+                (endPointAStr, endPointBStr) -> (endPointAStr + ',' + endPointBStr)
+        ).orElse("").getBytes(Charsets.UTF_8);
```

Comment:
How about indenting like this:

```java
        return endpointList.stream()
                           .map(endPointA -> (endPointA.authority() + ':' + endPointA.weight()))
                           .reduce((endPointAStr, endPointBStr) -> (endPointAStr + ',' + endPointBStr))
                           .orElse("").getBytes(Charsets.UTF_8);
```

Also, please remove redundant parehthesis in the lambda expressions.